File: src/my_project/easy_problems/from201to250/slowest_key.py
from typing import List 

class Solution:
    def slowestKey(self, releaseTimes: List[int], keysPressed: str) -> str:
        len_release_times: int = len(releaseTimes)
        releaseTimes = [0] + releaseTimes
        time_duration: int = list()
        dict_time_duration = dict()
        answer = list()
        
        for i in range(len_release_times):
            dict_time_duration[releaseTimes[i+1]-releaseTimes[i]] = []
        
        for i in range(len_release_times):
            time_duration.append(releaseTimes[i+1]-releaseTimes[i])
            dict_time_duration[releaseTimes[i+1]-releaseTimes[i]] += [keysPressed[i]]
            
        max_release_time: int = max(time_duration)
        
                  
        answer = dict_time_duration[max_release_time]
        answer.sort(reverse=True)          
        
        
        return answer[0]
    
solution = Solution()

print(solution.slowestKey(releaseTimes = [9,29,49,50], keysPressed = "cbcd"))


print(solution.slowestKey(releaseTimes = [12,23,36,46,62], keysPressed = "spuda"))

print(solution.slowestKey([19,22,28,29,66,81,93,97] ,"fnfaaxha"))

File: my_project_typescript/easy_problems_typescript/from101to150/slowestKey.ts
function slowestKey(releaseTimes: number[], keysPressed: string): string {

    let lenReleaseTimes: number = releaseTimes.length
    releaseTimes = [0].concat(releaseTimes)
    let timeDuration: Array<number> = []
    let dicTimeDuration: any = {}
    let answer: Array<string> = []

    for (let i = 0; i < lenReleaseTimes; i++){
        dicTimeDuration[releaseTimes[i+1] - releaseTimes[i]] = []
    }

    for (let i = 0; i < lenReleaseTimes; i++){
        timeDuration.push(releaseTimes[i+1] - releaseTimes[i])
        dicTimeDuration[releaseTimes[i+1] - releaseTimes[i]].push(keysPressed[i])
    }

    let maxReleaseTime: number = Math.max.apply(null, timeDuration)

    answer = dicTimeDuration[maxReleaseTime]
    answer.sort((a, b) =>  -a.localeCompare(b))

    return answer[0]
};


console.log(slowestKey([9,29,49,50], "cbcd"))

console.log(slowestKey([12,23,36,46,62], "spuda"))

console.log(slowestKey([19,22,28,29,66,81,93,97] ,"fnfaaxha"))
